src/aiodynamo/credentials.py

```diff
@@ -158,8 +158,9 @@ def __init__(
         self,
         *,
         path: Optional[Path] = None,
-        profile_name: str = "default",
+        profile_name: str = None,
     ) -> None:
+        profile_name = profile_name or os.environ.get("AWS_PROFILE", "default")
```

Comment:
I wonder if I understand this change correctly:
When credentials are in a file, the profile selector is an env var?
And when credentials are from env, there's no selector?

Reply:
That is correct. This is in line with the behavior of other AWS libraries such as boto3 or the AWS-CLI. The credential chain still respects the load order, and credentials set with AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, etc. or a profile passed explicitly when initializing the client will take precedence over the AWS_PROFILE env var.  

The behavior this mimics is described in these AWS docs:  
https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html
https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-profiles.html